### PR TITLE
Made Event and Session API responses timezone aware in ISO8601 format

### LINKED ISSUE #3575: Make Event and Session API responses timezone aware
Sub Issue of #3549 
I am working on this.

[Milestone 1 (GSoC 2017)](https://github.com/fossasia/open-event-orga-server/milestone/21)

### PR DESCRIPTION
**Pull request by mejariamol**
fixes #3575

### CODE REVIEW

File: app/models/event.py [hunk 1]
```diff
@@ -229,6 +231,39 @@ def get_staff_roles(self):
...
+            return None
+        return self.get_tz_aware_time(self.end_time)
+
+    @hybrid_property
+    def schedule_published_on_tz(self):
+        if not self.schedule_published_on:
+            return None
+        return self.get_tz_aware_time(self.schedule_published_on)
+
+    @start_time_tz.setter
```
Comment (mejariamol) [inline app/models/event.py]: @niranjan94 I have declared `setter` for hybrid properties. Shouldn't it manage POST requests. I am getting error in tests - 
```
File "/home/travis/build/fossasia/open-event-orga-server/app/api/helpers/helpers.py", line 235, in create_model

    new_model = model(**data)

TypeError: __init__() got an unexpected keyword argument 'start_time_tz'
```

**Comment by shubham-padia** [comment]
not sure if this is the right approach, changes in the custom field and changes in fix_payload should do the work, if we do not want to make any changes to how we store timezone in the database. @niranjan94 Suggestions please ?

**Comment by niranjan94** [comment]
I agree with @shubham-padia. This could be handled in a much simpler way as suggested by him. 

And yes, for now, we do not want to change how timezone is stored in the database.

**Comment by mejariamol** [comment]
@shubham-padia @niranjan94 will try on it.

File: app/models/event.py (hunk 1 above)
Comment (aviaryan) [inline app/models/event.py]: This error is simply because the Event model constructor (__init__) has no parameter named 'start_time_tz'.

**Comment by mejariamol** [comment]
@shubham-padia Just by changing `custom_fields.py` will not work, since datetime fields we getting from models are naive. `fix_payload` will do the thing for POST/PUT requests.
For GET requests I think we will need to define hybrid properties in the models. What do you think?

**Comment by shubham-padia** [comment]
@mejariamol for get requests, changes in the `format` method in the datetime class should do the work, AFAIK you can pass the event timezone as its attribute and localize the time accordingly in `format`. Please do tell if you face any problems.

**Comment by mejariamol** [comment]
@shubham-padia But time and timezone can't be passed separately to the datetime field class. It takes only one parameter as attribute. Thats why I was passing hybrid_property containing both.

**Comment by shubham-padia** [comment]
@mejariamol If you've confirmed that the said approach does not work, please proceed with your approach. Thanks !!

**Comment by shubham-padia** [comment]
` what do you say about this? how to pass both the infos (time and timezone) to the field.`

You can try passing a dict in the attribute, should work most probably.

**Comment by mejariamol** [comment]
@iamareebjamal No but. because attribute must be the field in the model AFAIK.

**Comment by mejariamol** [comment]
@niranjan94 @shubham-padia Shall we go with using `hybrid_property` and POST/PUT can be managed at `fix_payload` ..?

**Comment by shubham-padia** [comment]
@mejariamol If you're sure that that doesn't work, please go ahead :smile: .
@niranjan94 any comments ?

**Comment by niranjan94** [comment]
@shubham-padia @mejariamol no comments. You can go ahead with the `hybrid_property` implementation.

**Comment by mejariamol** [comment]
Outcome - 

1. GET responses of the event and session are timezone aware. `2016-06-06T11:22:33+0530`

2. POST/PUT/UPDATE requests follow same old pattern. i.e., `2016-06-06T11:22:33`

@niranjan94 @shubham-padia please review 
(current travis-ci error is not from the commit I guess.)

**Comment by shubham-padia** [comment]
@mejariamol The build is failing. Please look into that. :)

**Comment by mejariamol** [comment]
@shubham-padia yes. Currently I am AFK, will look into that as I get home. By tonight

**Comment by niranjan94** [comment]
@mejariamol what's the progress on this ? Please work with the rest of the team and try to get this out soon. Assigning it to Milestone 2

**Comment by mejariamol** [comment]
@SaptakS @shubham-padia @aviaryan I am kinda stuck here. Please take a look at travis build error. Test is failing at import event. KeyError is thrown for `background_image` while importing test - `/tests/unittests/api/test_import_more.py` which is untouched part by the commit.

Is it importing error ? #3585 
But why then test is not failing with other PR's if it is importing error. I couldn't figure out the bad in travis ci build.

**Comment by mejariamol** [comment]
Can be modified further once https://github.com/fossasia/open-event-orga-server/issues/3585 is resolved.

**Comment by mejariamol** [comment]
@niranjan94 I am getting this error on running tests on latest version-
```
IntegrityError: (psycopg2.IntegrityError) duplicate key value violates unique constraint "user_email_key"
DETAIL:  Key (email)=(test@example.com) already exists.
```
what does it mean?

**Comment by niranjan94** [comment]
It means you have a duplicate key. I think the error is pretty clear. 
Always run your tests with a clean database.

**Comment by mejariamol** [comment]
@niranjan94 yes . I ran `drop_db.py` before running tests.

**Comment by enigmaeth** [comment]
I think travis build would pass now as @shubham-padia added `test config` [here](https://github.com/fossasia/open-event-orga-server/pull/3625/files#diff-1f9f743421417fb2794c88447083a031R166).
If not, then as the error [here](https://travis-ci.org/fossasia/open-event-orga-server/builds/232314148#L2673) in travis build is Issue #3672  once PR #3675 is merged, the error might be resolved.

**Comment by mejariamol** [comment]
Yeah. I am waiting for PR #3675 to merge. I will recommit once the pr is merged.

**Comment by enigmaeth** [comment]
@mejariamol Please check now. The PR has been merged.

**Comment by mejariamol** [comment]
@niranjan94 @aviaryan @SaptakS  tests passed. Please review

Outcomes - 

1. GET responses of the event and session are timezone aware. `2016-06-06T11:22:33+05:30`

2. POST/PUT/UPDATE requests follow same old pattern. `2016-06-06T11:22:33`

**Comment by mejariamol** [comment]
@niranjan94 can be merged now.

**Pull request merged**